File: backend/routes/unitaMisura.js
const express = require('express');
const router = express.Router();
const db = require('../database');

router.get('/', (req, res) => {
  res.json(db.prepare('SELECT * FROM unita_misura ORDER BY nome').all());
});

router.post('/', (req, res) => {
  const { nome, simbolo } = req.body;
  const result = db.prepare('INSERT INTO unita_misura (nome, simbolo) VALUES (?,?)').run(nome?.trim(), simbolo?.trim() || nome?.trim());
  res.json({ id: result.lastInsertRowid });
});

router.put('/:id', (req, res) => {
  const { nome, simbolo } = req.body;
  db.prepare('UPDATE unita_misura SET nome=?, simbolo=? WHERE id=?').run(nome?.trim(), simbolo?.trim() || nome?.trim(), req.params.id);
  res.json({ success: true });
});

router.delete('/:id', (req, res) => {
  db.prepare('DELETE FROM unita_misura WHERE id=?').run(req.params.id);
  res.json({ success: true });
});

module.exports = router;

File: src-tauri/src/routes/unita_misura.rs
//! /api/unita-misura — parità con routes/unitaMisura.js

use axum::{
    extract::{Path, State},
    routing::{get, put},
    Json, Router,
};
use rusqlite::params;
use serde_json::{json, Value};

use crate::db::AppState;
use crate::error::ApiResult;
use crate::web::{str_field, str_or, tenant_conn};

pub fn routes() -> Router<AppState> {
    Router::new()
        .route("/", get(list).post(create))
        .route("/:id", put(update).delete(remove))
}

async fn list(State(state): State<AppState>) -> ApiResult<Json<Value>> {
    let conn = tenant_conn(&state)?;
    let conn = conn.lock().unwrap();
    let mut stmt = conn.prepare("SELECT id, nome, simbolo FROM unita_misura ORDER BY nome")?;
    let rows = stmt
        .query_map([], |r| {
            Ok(json!({
                "id": r.get::<_, i64>(0)?,
                "nome": r.get::<_, String>(1)?,
                "simbolo": r.get::<_, Option<String>>(2)?,
            }))
        })?
        .collect::<Result<Vec<_>, _>>()?;
    Ok(Json(Value::Array(rows)))
}

async fn create(State(state): State<AppState>, Json(body): Json<Value>) -> ApiResult<Json<Value>> {
    let nome = str_field(&body, "nome");
    let simbolo = str_or(&body, "simbolo", &nome);
    let conn = tenant_conn(&state)?;
    let conn = conn.lock().unwrap();
    conn.execute(
        "INSERT INTO unita_misura (nome, simbolo) VALUES (?1, ?2)",
        params![nome, simbolo],
    )?;
    Ok(Json(json!({ "id": conn.last_insert_rowid() })))
}

async fn update(
    State(state): State<AppState>,
    Path(id): Path<i64>,
    Json(body): Json<Value>,
) -> ApiResult<Json<Value>> {
    let nome = str_field(&body, "nome");
    let simbolo = str_or(&body, "simbolo", &nome);
    let conn = tenant_conn(&state)?;
    let conn = conn.lock().unwrap();
    conn.execute(
        "UPDATE unita_misura SET nome=?1, simbolo=?2 WHERE id=?3",
        params![nome, simbolo, id],
    )?;
    Ok(Json(json!({ "success": true })))
}

async fn remove(State(state): State<AppState>, Path(id): Path<i64>) -> ApiResult<Json<Value>> {
    let conn = tenant_conn(&state)?;
    let conn = conn.lock().unwrap();
    conn.execute("DELETE FROM unita_misura WHERE id=?1", params![id])?;
    Ok(Json(json!({ "success": true })))
}
